### PR TITLE
files cache, cacheD, EnsureRunning; closes #74, #113, #99 respectively

### LINKED ISSUE #74: `files cache`
make IPFS `pin` things. by hash

### PR DESCRIPTION
**Pull request by zramsay**

### CODE REVIEW

**Comment by zramsay** [comment]
left `get` and `put` through gateway.

File: util/writers.go [hunk 1]
```diff
@@ -17,8 +18,49 @@ func Untar(reader io.Reader, name, dest string) error {
 	return archive.Untar(reader, dest, &archive.TarOptions{NoLchown: true, Name: name})
 }
 
+//returns []byte to let each command make its own struct for the response
```
Comment (compleatang) [inline util/writers.go]: nice. good call.

File: util/readers.go [hunk 2]
```diff
@@ -19,11 +21,55 @@ func GetFromGithub(org, repo, branch, path, fileName string, w io.Writer) error
...
 
 func GetFromIPFS(hash, fileName string, w io.Writer) error {
-	url := IPFSBaseUrl() + hash
+	url := IPFSBaseUrl() + ":8080/ipfs/" + hash
 	w.Write([]byte("GETing file from IPFS. Hash =>\t" + hash + ":" + fileName + "\n"))
 	return DownloadFromUrlToFile(url, fileName, w)
 }
 
+func CatFromIPFS(fileHash string, w io.Writer) (string, error) {
+	url := IPFSBaseUrl() + ":5001/api/v0/cat?arg=" + fileHash
```
Comment (compleatang) [inline util/readers.go]: i think we should abstract the `:5001/api/v0/

File: util/readers.go [hunk 3]
```diff
@@ -19,11 +21,55 @@ func GetFromGithub(org, repo, branch, path, fileName string, w io.Writer) error
 }
 
 func GetFromIPFS(hash, fileName string, w io.Writer) error {
-	url := IPFSBaseUrl() + hash
+	url := IPFSBaseUrl() + ":8080/ipfs/" + hash
```
Comment (compleatang) [inline util/readers.go]: I think we should abstract the `:8080/ipfs/` into IPFSGatewayUrl() along similar lines to below.

File: commands/files.go [hunk 4]
```diff
@@ -41,6 +45,39 @@ var filesExport = &cobra.Command{
...
+	Use:   "cat [fileHash]",
+	Short: "Cat the contents of a file from IPFS.",
+	Long:  "Cat the contents of a file from IPFS.",
+	Run: func(cmd *cobra.Command, args []string) {
+		CatIt(cmd, args)
+	},
+}
+
+var filesList = &cobra.Command{
+	Use:   "ls [objectHash]",
```
Comment (compleatang) [inline commands/files.go]: does ipfs have a `this is what i have locally` functionality?

Comment (zramsay) [inline commands/files.go]: `eris files cached` now added

**Comment by compleatang** [comment]
LGTM. :walking: 

You ready to merge this?

**Pull request merged**